### ISSUE TITLE
Support for automatic use of delimiters (such as ```) to wrap user prompts.

### ISSUE THREAD

**Issue by tshu-w** [karthink/gptel#61]
According to https://learn.deeplearning.ai/chatgpt-prompt-eng/lesson/2/guidelines, we can use delimiters to make our input clearer. Delimiters can be anything like \`\`\`, """, `< >`, `<tag> </tag>`. Here is a motivating case:

I have a translator directive (system prompt):
```
(translator . "You are a professional translator. Translate the given text into Chinese if it is not written in Chinese, otherwise, translate it into English.")
```
If I send a text without delimiters, LLM may sometimes treat it as an instruction, here is an example:
```
Please explain the following content
```
the prompts are:
```
((:role "system" :content "You are a professional translator. Translate the given text into Chinese if it is not written in Chinese, otherwise translate it into English.") (:role "user" :content "Please explain the following content"))
```
and the reply is:
```
I'm sorry, but there is no specific content provided for me to explain. Could you please provide more context or information?
```
-----
If I wrap my text with delimiters like:
```
```\nPlease explain the following content\n```
```
now the prompts are:
```
((:role "system" :content "You are a professional translator. Translate the given text into Chinese if it is not written in Chinese, otherwise translate it into English.") (:role "user" :content "```
Please explain the following content
```"))
```
and the reply is:
```
请解释以下内容。
```

I believe this is helpful for non-dialogue (instructional) directives, including but not limited to writing documentation for code, debugging, summarizing, paraphrasing and so on.

**Comment by karthink** [comment]
Why not add the delimiters in the buffer around the text?

**Comment by tshu-w** [comment]
Because I want an automated way (instead of manually adding delimiters), for example, when writing articles, different segments need to be polished/translated/rewritten.
As mentioned above, I don't want the article content to affect the instructions.

**Comment by tshu-w** [comment]
I found that the latest models no longer tend to have the aforementioned issues (confusion between instructions and content), so not adding delimiters is also acceptable.

I'm closing this issue, but I still hope there will be an interface that allows users to customize request text (automatically adding delimiters) to accommodate different needs.

**Comment by karthink** [comment]
> I'm closing this issue, but I still hope there will be an interface that allows users to customize request text (automatically adding delimiters) to accommodate different needs.

Yes, I want to add a general input-filter hook.  Some other feature requests require it for other purposes, like replacing $(ls) with the output of the shell command.
